include/coap3/coap_session_internal.h

```diff
@@ -571,6 +571,33 @@ coap_session_t *coap_new_client_session_psk2_lkd(coap_context_t *ctx,
 coap_session_t *coap_session_new_dtls_session(coap_session_t *session,
                                               coap_tick_t now);
 
+/**
+ * Prints the names of all known resources for @p context to @p buf. This function
+ * sets @p buflen to the number of bytes actually written and returns
+ * @c 1 on success. On error, the value in @p buflen is undefined and
+ * the return value will be @c 0.
+ *
+ * @param context The context with the resource map.
+ * @param buf     The buffer to write the result.
+ * @param buflen  Must be initialized to the maximum length of @p buf and will be
+ *                set to the length of the well-known response on return.
+ * @param offset  The offset in bytes where the output shall start and is
+ *                shifted accordingly with the characters that have been
+ *                processed. This parameter is used to support the block
+ *                option.
+ * @param query_filter A filter query according to <a href="http://tools.ietf.org/html/draft-ietf-core-link-format-11#section-4.1">Link Format</a>
+ *
+ * @return COAP_PRINT_STATUS_ERROR on error. Otherwise, the lower 28 bits are
+ *         set to the number of bytes that have actually been written to
+ *         @p buf. COAP_PRINT_STATUS_TRUNC is set when the output has been
+ *         truncated.
+ */
+coap_print_status_t coap_print_wellknown_lkd(coap_context_t *context,
+                                             unsigned char *buf,
```

Comment:
This should be in` include/coap3/coap_resource_internal.h` for consistency with `coap_print_wellknown()` in `include/coap3/coap_resourse.h`.

Reply:
Ops.